medicines/search-client/src/document_type.rs

```diff
@@ -96,11 +166,38 @@ mod tests {
     #[test_case("\"Pil\"", DocumentType::Pil; "pil titlecase")]
     #[test_case("\"PAR\"", DocumentType::Par; "par uppercase")]
     #[test_case("\"Par\"", DocumentType::Par; "par titlecase")]
-    fn deserializes_cases_insensitively(input: &str, expected: DocumentType) {
+    fn deserializes_document_cases_insensitively(input: &str, expected: DocumentType) {
         use pretty_assertions::assert_eq;
 
         let from_str: DocumentType = serde_json::from_str(input).unwrap();
 
         assert_eq!(from_str, expected);
     }
+
+    #[test_case("UK")]
+    #[test_case("GB")]
+    #[test_case("NI")]
+    fn parses_territory_and_formats_to_a_string(territory_type: &str) {
+        use pretty_assertions::assert_eq;
+
+        let from_str: TerritoryType = territory_type.parse().unwrap();
+
+        let as_str = from_str.to_string();
+
+        assert_eq!(territory_type, as_str);
+    }
+
+    #[test_case("\"UK\"", TerritoryType::UK; "uk uppercase")]
+    #[test_case("\"Uk\"", TerritoryType::UK; "uk titlecase")]
+    #[test_case("\"GB\"", TerritoryType::GB; "gb uppercase")]
+    #[test_case("\"Gb\"", TerritoryType::GB; "gb titlecase")]
+    #[test_case("\"NI\"", TerritoryType::NI; "ni uppercase")]
+    #[test_case("\"Ni\"", TerritoryType::NI; "ni titlecase")]
```

Comment:
Guessing there are no lower-case requirements?

Reply:
AFAIK it's uppercase only, but i've added titlecase to be on par with how the `doc_type` is parsed. It's not outside the realms of possibility for it to be lowercase too, so I'll add that in.